provider/oci/oci.go

```diff
@@ -113,27 +115,61 @@ func (p *OCIProvider) zones(ctx context.Context) (map[string]dns.ZoneSummary, er
 
 	log.Debugf("Matching zones against domain filters: %v", p.domainFilter.Filters)
 	var page *string
-	for {
-		resp, err := p.client.ListZones(ctx, dns.ListZonesRequest{
-			CompartmentId: &p.cfg.CompartmentID,
-			ZoneType:      dns.ListZonesZoneTypePrimary,
-			Page:          page,
-		})
-		if err != nil {
-			return nil, errors.Wrapf(err, "listing zones in %q", p.cfg.CompartmentID)
-		}
 
-		for _, zone := range resp.Items {
-			if p.domainFilter.Match(*zone.Name) && p.zoneIDFilter.Match(*zone.Id) {
-				zones[*zone.Name] = zone
-				log.Debugf("Matched %q (%q)", *zone.Name, *zone.Id)
-			} else {
-				log.Debugf("Filtered %q (%q)", *zone.Name, *zone.Id)
+	// set zone scope based on configuration flag
+	scope := dns.ListZonesScopeEnum(p.zoneScope)
+	if scope == "" {
```

Comment:
I wonder if it's a better API to have scope == "ALL" instead of having scope "" imply both GLOBAL and PRIVATE?

Reply:
From user experience point of view, it should be `""` as AWS is also doing it this way (haven't analysed other providers though):
https://github.com/kubernetes-sigs/external-dns/blob/master/pkg/apis/externaldns/types.go#L395